src/jsdoc_transformer.ts

```diff
@@ -307,6 +288,55 @@ function createMemberTypeDeclaration(
   return ts.createIf(ts.createLiteral(false), ts.createBlock(propertyDecls, true));
 }
 
+/**
+ * Declares to Closure the supertypes of a given type.
+ * Normally you'd do this with \@implements or \@extends, but this is used in
+ * cases where we cannot express the types using those tags.
+ */
+function createSuperTypeRelationship(subType: string, superTypes: string[]): ts.IfStatement {
+  // For each supertype-subtype relationship, we create an unused function that does:
+  //   function tsickleSuperType(/** !Fruit */ supertype, /** !Apple */ subtype) {
+  //     supertype = subtype;
+  //   }
+  // This is an illegal assignment because the types aren't known to be related,
+  // but because checkTypes is off(!) it instead signals to the compiler that the
```

Comment:
I think even if it was on, it'd generate those signals - it being off just prevents the error from being emitted, but the internal optimizer model of JSCompiler still records the fact.